src/aics-image-viewer/components/App/index.tsx

```diff
@@ -292,21 +315,49 @@ const App: React.FC<AppProps> = (props) => {
       const oldRange = channelRangesRef.current[channelIndex];
       if (thisChannelsSettings.useControlPoints) {
         // control points were just automatically remapped - update in state
-        changeChannelSetting(channelIndex, "controlPoints", thisChannel.lut.controlPoints);
+        newControlPoints = thisChannel.lut.controlPoints;
         // now manually remap ramp using the channel's old range
-        const controlPoints = rampToControlPoints(thisChannelsSettings.ramp);
-        const newControlPoints = remapControlPointsForChannel(controlPoints, oldRange, thisChannel);
-        changeChannelSetting(channelIndex, "ramp", controlPointsToRamp(newControlPoints));
+        const rampControlPoints = rampToControlPoints(thisChannelsSettings.ramp);
+        const remappedRampControlPoints = remapControlPointsForChannel(rampControlPoints, oldRange, thisChannel);
+        newRamp = controlPointsToRamp(remappedRampControlPoints);
       } else {
         // ramp was just automatically remapped - update in state
-        const ramp = controlPointsToRamp(thisChannel.lut.controlPoints);
-        changeChannelSetting(channelIndex, "ramp", ramp);
+        newRamp = controlPointsToRamp(thisChannel.lut.controlPoints);
         // now manually remap control points using the channel's old range
-        const { controlPoints } = thisChannelsSettings;
-        const newControlPoints = remapControlPointsForChannel(controlPoints, oldRange, thisChannel);
-        changeChannelSetting(channelIndex, "controlPoints", newControlPoints);
+        const remappedControlPoints = remapControlPointsForChannel(
+          thisChannelsSettings.controlPoints,
+          oldRange,
+          thisChannel
+        );
+        newControlPoints = remappedControlPoints;
       }
     }
+    changeChannelSetting(channelIndex, "controlPoints", newControlPoints);
+    changeChannelSetting(channelIndex, "ramp", newRamp);
+
+    // Save the first loaded channel's volume subregion as our reference for resetting channel states.
+    if (!getSavedSubregionSize()) {
+      setSavedSubregionSize(aimg.imageInfo.subregionSize.clone());
+    }
+    // If we haven't saved the initial state for this channel yet and this is the same
+    // time and initial subregion, add the control points and ramp to the saved channel state.
+    const savedChannelState = getSavedChannelState(channelIndex);
+    if (
+      !savedChannelState &&
+      viewerSettings.time === aimg.loadSpec.time &&
+      matchesSavedSubregion(getSavedSubregionSize(), aimg.imageInfo.subregionSize)
+    ) {
+      const newState = {
+        ...thisChannelsSettings,
+        ramp: newRamp,
+        controlPoints: newControlPoints,
+        needsDefaultLut: false,
+      };
+      setSavedChannelState(channelIndex, newState);
+    }
+    // Callback to notify the viewer state that a channel has loaded.
+    onChannelLoaded(aimg, channelIndex);
```

Comment:
Debating whether this should be renamed; this callback actually gives the ResetStateProvider the chance to check for and reset the channel's control points and ramp if it's currently pending reset. Something like `checkForChannelResetOnLoad`?